// @coinbase/onchainkit/esm/nft/components/NFTMintCard.js
'use client';
import { useIsMounted } from '../../internal/hooks/useIsMounted.js';
import { useTheme } from '../../internal/hooks/useTheme.js';
import { NFTLifecycleProvider } from './NFTLifecycleProvider.js';
import { NFTProvider } from './NFTProvider.js';
import './mint/index.js';
import { useMintData } from '../hooks/useMintData.js';
import { LifecycleType } from '../types.js';
import { buildMintTransactionData } from '../utils/buildMintTransactionData.js';
import { cn, color, background, border } from '../../styles/theme.js';
import NFTErrorBoundary from './NFTErrorBoundary.js';
import { NFTErrorFallback } from './NFTErrorFallback.js';
import './view/index.js';
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { NFTCreator } from './mint/NFTCreator.js';
import { NFTMedia } from './view/NFTMedia.js';
import { NFTCollectionTitle } from './mint/NFTCollectionTitle.js';
import { NFTQuantitySelector } from './mint/NFTQuantitySelector.js';
import { NFTAssetCost } from './mint/NFTAssetCost.js';
import { NFTMintButton } from './mint/NFTMintButton.js';
function NFTMintCardDefaultContent() {
  return /*#__PURE__*/jsxs(Fragment, {
    children: [/*#__PURE__*/jsx(NFTCreator, {}), /*#__PURE__*/jsx(NFTMedia, {}), /*#__PURE__*/jsx(NFTCollectionTitle, {}), /*#__PURE__*/jsx(NFTQuantitySelector, {}), /*#__PURE__*/jsx(NFTAssetCost, {}), /*#__PURE__*/jsx(NFTMintButton, {})]
  });
}
function NFTMintCard({
  children = /*#__PURE__*/jsx(NFTMintCardDefaultContent, {}),
  className,
  contractAddress,
  tokenId,
  isSponsored,
  useNFTData = useMintData,
  buildMintTransaction = buildMintTransactionData,
  onStatus,
  onError,
  onSuccess
}) {
  const componentTheme = useTheme();
  const isMounted = useIsMounted();
  // prevents SSR hydration issue
  if (!isMounted) {
    return null;
  }
  return /*#__PURE__*/jsx(NFTErrorBoundary, {
    fallback: NFTErrorFallback,
    children: /*#__PURE__*/jsx(NFTLifecycleProvider, {
      type: LifecycleType.MINT,
      onStatus: onStatus,
      onError: onError,
      onSuccess: onSuccess,
      children: /*#__PURE__*/jsx(NFTProvider, {
        contractAddress: contractAddress,
        tokenId: tokenId,
        isSponsored: isSponsored,
        useNFTData: useNFTData,
        buildMintTransaction: buildMintTransaction,
        children: /*#__PURE__*/jsx("div", {
          className: cn(componentTheme, color.foreground, background.default, border.defaultActive, border.radius, 'flex w-full max-w-[500px] flex-col gap-2 border p-4', className),
          "data-testid": "ockNFTMintCard_Container",
          children: children
        })
      })
    })
  });
}
export { NFTMintCard };
