// @coinbase/onchainkit/esm/appchain/bridge/hooks/useAppchainConfig.js
'use client';
import { useReadContract } from 'wagmi';
import { DeployChainABI } from '../abi.js';
import { APPCHAIN_DEPLOY_CONTRACT_ADDRESS } from '../constants.js';
function useChainConfig(params) {
  const _useReadContract = useReadContract({
      abi: DeployChainABI,
      functionName: 'deployAddresses',
      args: [BigInt(params.appchainChainId)],
      address: APPCHAIN_DEPLOY_CONTRACT_ADDRESS[params.l2ChainId],
      query: {
        staleTime: 1000 * 60 * 60,
        // 1 hour
        retry: 2,
        enabled: !!params.l2ChainId && !!params.appchainChainId,
        gcTime: 0
      },
      // Read from the L2 contract
      chainId: params.l2ChainId
    }),
    data = _useReadContract.data,
    isLoading = _useReadContract.isLoading,
    isError = _useReadContract.isError,
    error = _useReadContract.error;
  return {
    config: error ? undefined : {
      chainId: params.appchainChainId,
      contracts: data
    },
    isLoading,
    isError,
    error
  };
}
export { useChainConfig };
