// @coinbase/onchainkit/esm/nft/components/mint/NFTQuantitySelector.js
import { QuantitySelector } from '../../../internal/components/QuantitySelector.js';
import { useNFTContext } from '../NFTProvider.js';
import { useMintAnalytics } from '../../hooks/useMintAnalytics.js';
import { cn } from '../../../styles/theme.js';
import { jsx } from 'react/jsx-runtime';
function NFTQuantitySelector({
  className
}) {
  const _useNFTContext = useNFTContext(),
    maxMintsPerWallet = _useNFTContext.maxMintsPerWallet,
    setQuantity = _useNFTContext.setQuantity;
  const _useMintAnalytics = useMintAnalytics(),
    handleQuantityChange = _useMintAnalytics.handleQuantityChange;

  // if max is 1, no need to show quantity selector
  if (maxMintsPerWallet === 1) {
    return null;
  }
  const handleQuantityUpdate = value => {
    setQuantity(value);
    const quantity = Number.parseInt(value, 10);
    if (!Number.isNaN(quantity)) {
      handleQuantityChange(quantity);
    }
  };
  return /*#__PURE__*/jsx("div", {
    className: cn('py-1', className),
    children: /*#__PURE__*/jsx(QuantitySelector, {
      className: className,
      onChange: handleQuantityUpdate,
      minQuantity: 1,
      maxQuantity: maxMintsPerWallet,
      placeholder: ""
    })
  });
}
export { NFTQuantitySelector };
