// @coinbase/onchainkit/esm/earn/components/VaultDetails.js
'use client';
function _slicedToArray(r, e) { return _arrayWithHoles(r) || _iterableToArrayLimit(r, e) || _unsupportedIterableToArray(r, e) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(r) { if (Array.isArray(r)) return r; }
import { getTruncatedAmount } from '../utils/getTruncatedAmount.js';
import { Popover } from '../../internal/components/Popover.js';
import { Skeleton } from '../../internal/components/Skeleton.js';
import { etherscanSvg } from '../../internal/svg/etherscanSvg.js';
import { infoSvg } from '../../internal/svg/infoSvg.js';
import { cn, color, background, border, text } from '../../styles/theme.js';
import '../../token/index.js';
import { useState, useRef } from 'react';
import { useEarnContext } from './EarnProvider.js';
import { jsxs, jsx } from 'react/jsx-runtime';
import { TokenImage } from '../../token/components/TokenImage.js';
function Row({
  label,
  value
}) {
  return /*#__PURE__*/jsxs("div", {
    className: "flex items-center justify-between gap-8",
    children: [/*#__PURE__*/jsx("div", {
      children: label
    }), /*#__PURE__*/jsx("div", {
      className: "font-semibold",
      children: value
    })]
  });
}
function VaultDetails() {
  const _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    isOpen = _useState2[0],
    setIsOpen = _useState2[1];
  const triggerRef = useRef(null);
  const anchorRef = useRef(null);
  const _useEarnContext = useEarnContext(),
    vaultToken = _useEarnContext.vaultToken,
    vaultName = _useEarnContext.vaultName,
    deposits = _useEarnContext.deposits,
    liquidity = _useEarnContext.liquidity,
    vaultAddress = _useEarnContext.vaultAddress;
  if (!vaultToken || !vaultName) {
    return /*#__PURE__*/jsx(Skeleton, {
      className: "!rounded-full h-7 min-w-28"
    });
  }
  return /*#__PURE__*/jsxs("div", {
    ref: anchorRef,
    className: cn(text.label1, color.foregroundMuted, background.alternate, 'flex items-center justify-center gap-2 rounded-full p-1 px-3'),
    "data-testid": "ock-vaultDetails",
    children: [/*#__PURE__*/jsx(TokenImage, {
      token: vaultToken,
      size: 16
    }), /*#__PURE__*/jsx("span", {
      className: "max-w-24 truncate",
      title: vaultName,
      children: vaultName
    }), /*#__PURE__*/jsx("button", {
      ref: triggerRef,
      type: "button",
      "data-testid": "ock-vaultDetailsButton",
      className: cn('size-3 [&_path]:fill-[var(--ock-icon-color-foreground-muted)] [&_path]:transition-colors [&_path]:ease-in-out [&_path]:hover:fill-[var(--ock-icon-color-foreground)]', isOpen && '[&_path]:fill-[var(--ock-icon-color-foreground)]'),
      onClick: () => setIsOpen(!isOpen),
      children: infoSvg
    }), /*#__PURE__*/jsx(Popover, {
      isOpen: isOpen,
      onClose: () => setIsOpen(false),
      position: "bottom",
      align: "start",
      trigger: triggerRef,
      anchor: anchorRef.current,
      offset: 4,
      children: /*#__PURE__*/jsxs("div", {
        className: cn(background.default, color.foreground, border.defaultActive, background.default, 'flex min-w-40 flex-col gap-3 rounded-lg border p-3 text-sm', 'fade-in animate-in duration-200'),
        children: [/*#__PURE__*/jsx("div", {
          className: "font-semibold",
          children: vaultName
        }), /*#__PURE__*/jsxs("div", {
          className: "flex flex-col gap-2",
          children: [/*#__PURE__*/jsx(Row, {
            label: "Token",
            value: /*#__PURE__*/jsxs("div", {
              className: "flex items-center gap-1",
              children: [/*#__PURE__*/jsx(TokenImage, {
                token: vaultToken,
                size: 16
              }), /*#__PURE__*/jsx("span", {
                className: "max-w-24 truncate",
                children: vaultToken.symbol
              })]
            })
          }), deposits ? /*#__PURE__*/jsx(Row, {
            label: "Total deposits",
            value: `${getTruncatedAmount(deposits, 1, 'compact')} ${vaultToken.symbol}`
          }) : null, liquidity ? /*#__PURE__*/jsx(Row, {
            label: "Liquidity",
            value: `${getTruncatedAmount(liquidity, 1, 'compact')} ${vaultToken.symbol}`
          }) : null]
        }), /*#__PURE__*/jsxs("a", {
          href: `https://basescan.org/address/${vaultAddress}`,
          target: "_blank",
          rel: "noopener noreferrer",
          className: cn(color.primary, 'flex max-w-fit items-center gap-1 hover:opacity-80'),
          "data-testid": "ock-vaultDetailsBaseScanLink",
          children: [/*#__PURE__*/jsx("div", {
            children: "View on BaseScan"
          }), /*#__PURE__*/jsx("div", {
            className: "h-3 w-3",
            children: etherscanSvg
          })]
        })]
      })
    })]
  });
}
export { VaultDetails };
