// @coinbase/onchainkit/esm/earn/components/WithdrawBalance.js
'use client';
import { getTruncatedAmount } from '../utils/getTruncatedAmount.js';
import { Skeleton } from '../../internal/components/Skeleton.js';
import { useCallback, useMemo } from 'react';
import { useAccount } from 'wagmi';
import { EarnBalance } from './EarnBalance.js';
import { useEarnContext } from './EarnProvider.js';
import { jsx, jsxs } from 'react/jsx-runtime';
function WithdrawBalance({
  className
}) {
  const _useEarnContext = useEarnContext(),
    depositedBalance = _useEarnContext.depositedBalance,
    status = _useEarnContext.depositedBalanceStatus,
    setWithdrawAmount = _useEarnContext.setWithdrawAmount,
    vaultToken = _useEarnContext.vaultToken;
  const _useAccount = useAccount(),
    address = _useAccount.address;
  const handleMaxPress = useCallback(() => {
    if (depositedBalance) {
      setWithdrawAmount(depositedBalance);
    }
  }, [depositedBalance, setWithdrawAmount]);
  const balance = useMemo(() => {
    if (!depositedBalance) {
      return '0';
    }
    return getTruncatedAmount(depositedBalance.toString(), 4);
  }, [depositedBalance]);
  const title = useMemo(() => {
    if (!address) {
      return 'Wallet not connected';
    }
    if (!vaultToken) {
      return /*#__PURE__*/jsx(Skeleton, {
        className: "h-6 w-24"
      });
    }
    if (status === 'pending') {
      return /*#__PURE__*/jsxs("div", {
        className: "flex gap-1",
        children: [/*#__PURE__*/jsx(Skeleton, {
          className: "!bg-[var(--ock-bg-alternate-active)] h-6 w-12"
        }), /*#__PURE__*/jsx("span", {
          children: vaultToken?.symbol
        })]
      });
    }
    return `${balance} ${vaultToken?.symbol}`;
  }, [balance, vaultToken, address, status]);
  const subtitle = useMemo(() => {
    if (!address) {
      return 'Connect wallet to withdraw';
    }
    return 'Available to withdraw';
  }, [address]);
  return /*#__PURE__*/jsx(EarnBalance, {
    className: className,
    title: title,
    subtitle: subtitle,
    onActionPress: handleMaxPress,
    showAction: balance !== '0'
  });
}
export { WithdrawBalance };
