// @coinbase/onchainkit/esm/earn/components/YieldDetails.js
'use client';
function _slicedToArray(r, e) { return _arrayWithHoles(r) || _iterableToArrayLimit(r, e) || _unsupportedIterableToArray(r, e) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(r) { if (Array.isArray(r)) return r; }
import { useEarnContext } from './EarnProvider.js';
import { getTruncatedAmount } from '../utils/getTruncatedAmount.js';
import { Popover } from '../../internal/components/Popover.js';
import { Skeleton } from '../../internal/components/Skeleton.js';
import { infoSvg } from '../../internal/svg/infoSvg.js';
import { formatPercent } from '../../internal/utils/formatPercent.js';
import { cn, color, border, background, text } from '../../styles/theme.js';
import { useState, useRef } from 'react';
import { jsxs, jsx } from 'react/jsx-runtime';
function YieldInfo() {
  const _useEarnContext = useEarnContext(),
    rewards = _useEarnContext.rewards,
    nativeApy = _useEarnContext.nativeApy,
    vaultToken = _useEarnContext.vaultToken,
    vaultFee = _useEarnContext.vaultFee;
  return /*#__PURE__*/jsxs("div", {
    className: cn(color.foregroundMuted, border.defaultActive, background.default, 'fade-in flex min-w-52 animate-in flex-col gap-2 rounded-lg border p-3 text-sm duration-200'),
    children: [nativeApy ? /*#__PURE__*/jsxs("div", {
      className: "flex items-center justify-between gap-1",
      "data-testid": "ock-earnNativeApy",
      children: [/*#__PURE__*/jsx("div", {
        children: vaultToken?.symbol
      }), /*#__PURE__*/jsx("div", {
        className: "font-semibold",
        children: formatPercent(nativeApy)
      })]
    }) : null, rewards?.map(reward => /*#__PURE__*/jsxs("div", {
      className: "flex items-center justify-between gap-1",
      "data-testid": "ock-earnRewards",
      children: [/*#__PURE__*/jsx("div", {
        children: reward.assetName
      }), /*#__PURE__*/jsx("div", {
        className: "font-semibold",
        children: formatPercent(reward.apy)
      })]
    }, reward.asset)), vaultFee && nativeApy ? /*#__PURE__*/jsxs("div", {
      className: "flex items-center justify-between gap-1",
      "data-testid": "ock-earnPerformanceFee",
      children: [/*#__PURE__*/jsxs("div", {
        children: ["Perf. Fee", ' ', /*#__PURE__*/jsxs("span", {
          className: "text-xs",
          children: ["(", formatPercent(vaultFee, 0), ")"]
        })]
      }), /*#__PURE__*/jsxs("div", {
        className: "font-semibold",
        children: ["-", formatPercent(vaultFee * nativeApy)]
      })]
    }) : null]
  });
}
function YieldDetails() {
  const _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    isOpen = _useState2[0],
    setIsOpen = _useState2[1];
  const triggerRef = useRef(null);
  const anchorRef = useRef(null);
  const _useEarnContext2 = useEarnContext(),
    apy = _useEarnContext2.apy;
  if (!apy) {
    return /*#__PURE__*/jsx(Skeleton, {
      className: "!rounded-full h-7 min-w-28"
    });
  }
  return /*#__PURE__*/jsxs("div", {
    ref: anchorRef,
    className: cn(text.label1, color.foregroundMuted, background.alternate, 'flex items-center justify-center gap-1 rounded-full p-1 px-3'),
    "data-testid": "ock-yieldDetails",
    children: [`APY ${formatPercent(Number(getTruncatedAmount(apy.toString(), 4)))}`, /*#__PURE__*/jsx("button", {
      ref: triggerRef,
      type: "button",
      "data-testid": "ock-apyInfoButton",
      className: cn('size-3 [&_path]:fill-[var(--ock-icon-color-foreground-muted)] [&_path]:transition-colors [&_path]:ease-in-out [&_path]:hover:fill-[var(--ock-icon-color-foreground)]', isOpen && '[&_path]:fill-[var(--ock-icon-color-foreground)]'),
      onClick: () => setIsOpen(!isOpen),
      children: infoSvg
    }), /*#__PURE__*/jsx(Popover, {
      isOpen: isOpen,
      onClose: () => setIsOpen(false),
      position: "bottom",
      align: "end",
      trigger: triggerRef,
      anchor: anchorRef.current,
      offset: 4,
      children: /*#__PURE__*/jsx(YieldInfo, {})
    })]
  });
}
export { YieldDetails };
