// @coinbase/onchainkit/esm/token/components/TokenBalance.js
const _excluded = ["token", "onClick", "onActionPress", "actionText", "classNames", "aria-label"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
import { formatFiatAmount } from '../../internal/utils/formatFiatAmount.js';
import { truncateDecimalPlaces } from '../../internal/utils/truncateDecimalPlaces.js';
import { cn, text, color, border } from '../../styles/theme.js';
import { TokenImage } from './TokenImage.js';
import { useMemo } from 'react';
import { formatUnits } from 'viem';
import { jsxs, jsx } from 'react/jsx-runtime';
function TokenBalance(_ref) {
  let token = _ref.token,
    onClick = _ref.onClick,
    onActionPress = _ref.onActionPress,
    _ref$actionText = _ref.actionText,
    actionText = _ref$actionText === void 0 ? 'Max' : _ref$actionText,
    classNames = _ref.classNames,
    ariaLabel = _ref['aria-label'],
    contentProps = _objectWithoutProperties(_ref, _excluded);
  if (onClick) {
    return /*#__PURE__*/jsxs("div", {
      className: "relative",
      children: [/*#__PURE__*/jsx("button", {
        type: "button",
        "aria-label": ariaLabel ?? `${token.name} token balance`,
        onClick: () => onClick(token),
        className: cn('flex w-full items-center justify-start gap-4 px-2 py-1', classNames?.container),
        "data-testid": "ockTokenBalanceButton",
        children: /*#__PURE__*/jsx(TokenBalanceContent, _objectSpread({
          token: token,
          classNames: classNames,
          onActionPress: onActionPress
        }, contentProps))
      }), onActionPress && /*#__PURE__*/jsx(ActionButton, {
        actionText: actionText,
        onActionPress: onActionPress,
        className: classNames?.action
      })]
    });
  }
  return /*#__PURE__*/jsxs("div", {
    className: cn('flex w-full items-center justify-start gap-4 px-2 py-1', classNames?.container),
    "data-testid": "ockTokenBalance",
    children: [/*#__PURE__*/jsx(TokenBalanceContent, _objectSpread(_objectSpread({
      token: token
    }, contentProps), {}, {
      classNames: classNames
    })), onActionPress && /*#__PURE__*/jsx(ActionButton, {
      actionText: actionText,
      onActionPress: onActionPress,
      className: classNames?.action
    })]
  });
}
function TokenBalanceContent({
  token,
  subtitle,
  showImage = true,
  onActionPress,
  tokenSize = 40,
  classNames
}) {
  const formattedFiatValue = useMemo(() => formatFiatAmount({
    amount: token.fiatBalance,
    currency: 'USD'
  }), [token.fiatBalance]);
  const formattedCryptoValue = useMemo(() => truncateDecimalPlaces(formatUnits(BigInt(token.cryptoBalance), token.decimals), 3), [token.cryptoBalance, token.decimals]);
  return /*#__PURE__*/jsxs("div", {
    className: "grid w-full grid-cols-[2.5rem_1fr_auto] items-center gap-3",
    children: [/*#__PURE__*/jsx("div", {
      className: "h-10 w-10",
      children: showImage && /*#__PURE__*/jsx(TokenImage, {
        token: token,
        size: tokenSize
      })
    }), /*#__PURE__*/jsxs("div", {
      className: "flex min-w-0 flex-col text-left",
      children: [/*#__PURE__*/jsx("span", {
        className: cn(text.headline, color.foreground, 'overflow-hidden text-ellipsis whitespace-nowrap', classNames?.tokenName),
        children: token.name?.trim()
      }), /*#__PURE__*/jsx("span", {
        className: cn(text.label2, color.foregroundMuted, classNames?.tokenValue),
        children: `${formattedCryptoValue} ${token.symbol} ${subtitle ?? ''}`
      })]
    }), /*#__PURE__*/jsx("div", {
      className: "text-right",
      children: !onActionPress && /*#__PURE__*/jsx("span", {
        className: cn(text.label2, color.foregroundMuted, 'whitespace-nowrap', classNames?.fiatValue),
        children: formattedFiatValue
      })
    })]
  });
}
function ActionButton({
  actionText,
  onActionPress,
  className
}) {
  return /*#__PURE__*/jsx("button", {
    type: "button",
    "data-testid": "ockTokenBalanceAction",
    "aria-label": actionText,
    onClick: onActionPress,
    className: cn(text.label2, color.primary, border.radius, 'cursor-pointer p-0.5 font-bold', 'border border-transparent hover:border-[--ock-line-primary]', '-translate-y-1/2 absolute top-1/2 right-2', className),
    children: actionText
  });
}
export { TokenBalance };
