// @coinbase/onchainkit/esm/appchain/bridge/components/AppchainBridgeResumeTransaction.js
'use client';
function _slicedToArray(r, e) { return _arrayWithHoles(r) || _iterableToArrayLimit(r, e) || _unsupportedIterableToArray(r, e) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(r) { if (Array.isArray(r)) return r; }
import { PressableIcon } from '../../../internal/components/PressableIcon.js';
import { TextInput } from '../../../internal/components/TextInput.js';
import { backArrowSvg } from '../../../internal/svg/backArrowSvg.js';
import { cn, text, color, background, border, pressable } from '../../../styles/theme.js';
import { useState } from 'react';
import { useAppchainBridgeContext } from './AppchainBridgeProvider.js';
import { jsx, jsxs } from 'react/jsx-runtime';
const AppchainBridgeResumeTransaction = () => {
  const _useAppchainBridgeCon = useAppchainBridgeContext(),
    setIsResumeTransactionModalOpen = _useAppchainBridgeCon.setIsResumeTransactionModalOpen,
    handleResumeTransaction = _useAppchainBridgeCon.handleResumeTransaction;
  const _useState = useState(null),
    _useState2 = _slicedToArray(_useState, 2),
    withdrawalTxHash = _useState2[0],
    setWithdrawalTxHash = _useState2[1];
  const _useState3 = useState(false),
    _useState4 = _slicedToArray(_useState3, 2),
    invalidInput = _useState4[0],
    setInvalidInput = _useState4[1];
  const backButton = /*#__PURE__*/jsx(PressableIcon, {
    ariaLabel: "Back button",
    onClick: () => {
      setIsResumeTransactionModalOpen(false);
    },
    children: /*#__PURE__*/jsx("div", {
      className: "p-2",
      children: backArrowSvg
    })
  });
  return /*#__PURE__*/jsxs("div", {
    className: "flex h-full w-full flex-col justify-between",
    children: [/*#__PURE__*/jsxs("div", {
      children: [/*#__PURE__*/jsxs("div", {
        className: "flex items-center",
        children: [backButton, /*#__PURE__*/jsx("h2", {
          className: "ock-text-foreground flex-1 text-center font-medium text-md",
          children: "Resume Transaction"
        })]
      }), /*#__PURE__*/jsxs("div", {
        className: cn(background.secondary, border.radius, 'box-border flex h-20 w-full flex-col items-start justify-center gap-2 p-4', 'mt-4'),
        children: [/*#__PURE__*/jsx("span", {
          className: cn(text.label2, color.foregroundMuted, 'flex items-center gap-1'),
          children: "Transaction hash"
        }), /*#__PURE__*/jsx(TextInput, {
          className: cn(text.label2, color.foregroundMuted, background.secondary, 'w-full border-none', 'focus:border-none focus:outline-none focus:ring-0'),
          placeholder: "0x...",
          onChange: value => {
            setWithdrawalTxHash(value);
          },
          value: withdrawalTxHash || ''
        })]
      }), withdrawalTxHash && invalidInput && /*#__PURE__*/jsx("div", {
        className: "mt-2 flex",
        children: /*#__PURE__*/jsx("p", {
          className: "text-foregroundMuted text-red-500 text-sm",
          children: "Please enter a valid transaction hash"
        })
      })]
    }), /*#__PURE__*/jsx("div", {
      className: "flex w-full justify-between",
      children: /*#__PURE__*/jsx("button", {
        type: "button",
        className: cn(pressable.primary, 'w-full rounded-xl px-4 py-3 font-medium text-base leading-6', 'text-center'),
        onClick: () => {
          if (!withdrawalTxHash?.startsWith('0x') || withdrawalTxHash.length !== 66) {
            setInvalidInput(true);
            return;
          }
          handleResumeTransaction(withdrawalTxHash);
        },
        children: /*#__PURE__*/jsx("div", {
          className: cn(text.headline, color.inverse, 'flex justify-center'),
          children: "Resume Transaction"
        })
      })
    })]
  });
};
export { AppchainBridgeResumeTransaction };
