// @coinbase/onchainkit/esm/wallet/components/ConnectWallet.js
'use client';
function _slicedToArray(r, e) { return _arrayWithHoles(r) || _iterableToArrayLimit(r, e) || _unsupportedIterableToArray(r, e) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(r) { if (Array.isArray(r)) return r; }
import '../../identity/index.js';
import { useState, useMemo, Children, isValidElement, useCallback, useEffect } from 'react';
import { useAccount, useConnect } from 'wagmi';
import { useAnalytics } from '../../core/analytics/hooks/useAnalytics.js';
import { WalletEvent } from '../../core/analytics/types.js';
import { IdentityProvider } from '../../identity/components/IdentityProvider.js';
import { Spinner } from '../../internal/components/Spinner.js';
import { findComponent } from '../../internal/utils/findComponent.js';
import { cn, pressable, text, color, border } from '../../styles/theme.js';
import { useOnchainKit } from '../../useOnchainKit.js';
import { ConnectButton } from './ConnectButton.js';
import { ConnectWalletText } from './ConnectWalletText.js';
import { WalletModal } from './WalletModal.js';
import { useWalletContext } from './WalletProvider.js';
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { Avatar } from '../../identity/components/Avatar.js';
import { Name } from '../../identity/components/Name.js';
const connectWalletDefaultChildren = /*#__PURE__*/jsxs(Fragment, {
  children: [/*#__PURE__*/jsx(Avatar, {
    className: "h-6 w-6"
  }), /*#__PURE__*/jsx(Name, {})]
});
function ConnectWallet({
  children,
  className,
  // In a few version we will officially deprecate this prop,
  // but for now we will keep it for backward compatibility.
  text: text$1 = 'Connect Wallet',
  onConnect,
  disconnectedLabel
}) {
  const _useOnchainKit = useOnchainKit(),
    _useOnchainKit$config = _useOnchainKit.config,
    config = _useOnchainKit$config === void 0 ? {
      wallet: {
        display: undefined
      }
    } : _useOnchainKit$config;
  // Core Hooks
  const _useWalletContext = useWalletContext(),
    setIsConnectModalOpen = _useWalletContext.setIsConnectModalOpen,
    isSubComponentOpen = _useWalletContext.isSubComponentOpen,
    setIsSubComponentOpen = _useWalletContext.setIsSubComponentOpen,
    handleClose = _useWalletContext.handleClose;
  const _useAccount = useAccount(),
    accountAddress = _useAccount.address,
    status = _useAccount.status,
    accountConnector = _useAccount.connector;
  const _useConnect = useConnect(),
    connectors = _useConnect.connectors,
    connect = _useConnect.connect,
    connectStatus = _useConnect.status;
  const _useAnalytics = useAnalytics(),
    sendAnalytics = _useAnalytics.sendAnalytics;
  // State
  const _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    hasClickedConnect = _useState2[0],
    setHasClickedConnect = _useState2[1];
  const _useState3 = useState(false),
    _useState4 = _slicedToArray(_useState3, 2),
    isModalOpen = _useState4[0],
    setIsModalOpen = _useState4[1]; // duplicate modal state because ConnectWallet not always within WalletProvider
  // TODO: remove lines 57-74 after deprecating ConnectWalletText
  // Get connectWalletText from children when present,
  // this is used to customize the connect wallet button text
  const _useMemo = useMemo(() => {
      const childrenArray = Children.toArray(children);
      return {
        connectWalletText: childrenArray.find(findComponent(ConnectWalletText))
      };
    }, [children]),
    connectWalletText = _useMemo.connectWalletText;
  // Remove connectWalletText from children if present
  const childrenWithoutConnectWalletText = useMemo(() => {
    return Children.map(children, child => {
      if (/*#__PURE__*/isValidElement(child) && child.type === ConnectWalletText) {
        return null;
      }
      return child;
    });
  }, [children]);
  // Wallet connect status
  const connector = accountConnector || connectors[0];
  const isLoading = connectStatus === 'pending' || status === 'connecting';
  // Handles
  const handleToggle = useCallback(() => {
    if (isSubComponentOpen) {
      handleClose?.(); // optional because ConnectWallet not always within WalletProvider
    } else {
      setIsSubComponentOpen(true);
    }
  }, [isSubComponentOpen, handleClose, setIsSubComponentOpen]);
  const handleCloseConnectModal = useCallback(() => {
    setIsModalOpen(false); // duplicate state because ConnectWallet not always within WalletProvider
    setIsConnectModalOpen?.(false); // optional because ConnectWallet not always within WalletProvider
  }, [setIsConnectModalOpen]);
  const handleOpenConnectModal = useCallback(() => {
    setIsModalOpen(true); // duplicate state because ConnectWallet not always within WalletProvider
    setIsConnectModalOpen?.(true); // optional because ConnectWallet not always within WalletProvider
    setHasClickedConnect(true);
  }, [setIsConnectModalOpen]);
  const handleAnalyticsInitiated = useCallback(component => {
    sendAnalytics(WalletEvent.ConnectInitiated, {
      component
    });
  }, [sendAnalytics]);
  const handleAnalyticsSuccess = useCallback(walletAddress => {
    const walletProvider = connector?.name;
    sendAnalytics(WalletEvent.ConnectSuccess, {
      address: walletAddress ?? '',
      walletProvider
    });
  }, [sendAnalytics, connector]);
  const handleAnalyticsError = useCallback((errorMessage, component) => {
    const walletProvider = connector?.name;
    sendAnalytics(WalletEvent.ConnectError, {
      error: errorMessage,
      metadata: {
        connector: walletProvider,
        component
      }
    });
  }, [sendAnalytics, connector]);
  // Effects
  useEffect(() => {
    if (hasClickedConnect && status === 'connected' && onConnect) {
      onConnect();
      setHasClickedConnect(false);
    }
  }, [status, hasClickedConnect, onConnect]);
  useEffect(() => {
    if (status === 'connected' && accountAddress && connector) {
      handleAnalyticsSuccess(accountAddress);
    }
  }, [status, accountAddress, connector, handleAnalyticsSuccess]);
  const handleConnectClick = useCallback(() => {
    if (config?.wallet?.display === 'modal') {
      handleOpenConnectModal();
      setHasClickedConnect(true);
      handleAnalyticsInitiated('WalletModal');
      return;
    }
    handleAnalyticsInitiated('ConnectWallet');
    connect({
      connector
    }, {
      onSuccess: () => {
        onConnect?.();
        handleAnalyticsSuccess(accountAddress);
      },
      onError: error => {
        handleAnalyticsError(error.message, 'ConnectWallet');
      }
    });
  }, [config?.wallet?.display, accountAddress, connect, connector, handleAnalyticsError, handleAnalyticsInitiated, handleAnalyticsSuccess, handleOpenConnectModal, onConnect]);
  if (status === 'disconnected') {
    return /*#__PURE__*/jsxs("div", {
      className: "flex",
      "data-testid": "ockConnectWallet_Container",
      children: [/*#__PURE__*/jsx(ConnectButton, {
        className: className,
        connectWalletText: connectWalletText || disconnectedLabel,
        onClick: handleConnectClick,
        text: text$1
      }), config?.wallet?.display === 'modal' && /*#__PURE__*/jsx(WalletModal, {
        isOpen: isModalOpen,
        onClose: handleCloseConnectModal
      })]
    });
  }
  if (isLoading) {
    return /*#__PURE__*/jsx("div", {
      className: "flex",
      "data-testid": "ockConnectWallet_Container",
      children: /*#__PURE__*/jsx("button", {
        type: "button",
        "data-testid": "ockConnectAccountButtonInner",
        className: cn(pressable.primary, text.headline, color.inverse, 'inline-flex min-w-[153px] items-center justify-center rounded-xl px-4 py-3', pressable.disabled, className),
        disabled: true,
        children: /*#__PURE__*/jsx(Spinner, {})
      })
    });
  }
  return /*#__PURE__*/jsx(IdentityProvider, {
    address: accountAddress,
    children: /*#__PURE__*/jsx("div", {
      className: "flex gap-4",
      "data-testid": "ockConnectWallet_Container",
      children: /*#__PURE__*/jsx("button", {
        type: "button",
        "data-testid": "ockConnectWallet_Connected",
        className: cn(pressable.secondary, border.radius, color.foreground, 'px-4 py-3', isSubComponentOpen && 'ock-bg-secondary-active hover:ock-bg-secondary-active', className),
        onClick: handleToggle,
        children: /*#__PURE__*/jsx("div", {
          className: "flex items-center justify-center gap-2",
          children: childrenWithoutConnectWalletText || connectWalletDefaultChildren
        })
      })
    })
  });
}
export { ConnectWallet };
