// @coinbase/onchainkit/esm/wallet/components/WalletAdvancedSwap.js
'use client';
import { PressableIcon } from '../../internal/components/PressableIcon.js';
import { backArrowSvg } from '../../internal/svg/backArrowSvg.js';
import { cn, border } from '../../styles/theme.js';
import { Swap } from '../../swap/components/Swap.js';
import { SwapAmountInput } from '../../swap/components/SwapAmountInput.js';
import { SwapButton } from '../../swap/components/SwapButton.js';
import { SwapMessage } from '../../swap/components/SwapMessage.js';
import { SwapSettings } from '../../swap/components/SwapSettings.js';
import { SwapSettingsSlippageDescription } from '../../swap/components/SwapSettingsSlippageDescription.js';
import { SwapSettingsSlippageInput } from '../../swap/components/SwapSettingsSlippageInput.js';
import { SwapSettingsSlippageTitle } from '../../swap/components/SwapSettingsSlippageTitle.js';
import { SwapToast } from '../../swap/components/SwapToast.js';
import { SwapToggleButton } from '../../swap/components/SwapToggleButton.js';
import { useCallback } from 'react';
import { useWalletContext } from './WalletProvider.js';
import { jsx, jsxs } from 'react/jsx-runtime';
function WalletAdvancedSwap({
  config,
  classNames,
  disabled,
  experimental,
  from,
  isSponsored = false,
  onError,
  onStatus,
  onSuccess,
  title,
  to
}) {
  const _useWalletContext = useWalletContext(),
    setActiveFeature = _useWalletContext.setActiveFeature,
    isActiveFeatureClosing = _useWalletContext.isActiveFeatureClosing,
    setIsActiveFeatureClosing = _useWalletContext.setIsActiveFeatureClosing;
  const handleCloseSwap = useCallback(() => {
    setIsActiveFeatureClosing(true);
  }, [setIsActiveFeatureClosing]);
  const handleAnimationEnd = useCallback(() => {
    if (isActiveFeatureClosing) {
      setActiveFeature(null);
      setIsActiveFeatureClosing(false);
    }
  }, [isActiveFeatureClosing, setActiveFeature, setIsActiveFeatureClosing]);
  const backButton = /*#__PURE__*/jsx(PressableIcon, {
    ariaLabel: "Back button",
    onClick: handleCloseSwap,
    children: /*#__PURE__*/jsx("div", {
      className: "p-2",
      children: backArrowSvg
    })
  });
  return /*#__PURE__*/jsx("div", {
    className: cn('h-full', border.radius, isActiveFeatureClosing ? 'fade-out slide-out-to-right-5 animate-out fill-mode-forwards ease-in-out' : 'fade-in slide-in-from-right-5 linear animate-in duration-150', 'relative w-88 h-120', classNames?.container),
    onAnimationEnd: handleAnimationEnd,
    "data-testid": "ockWalletAdvancedSwap",
    children: /*#__PURE__*/jsxs(Swap, {
      className: cn('w-full px-4 pt-3 pb-4', classNames?.container),
      onStatus: onStatus,
      onSuccess: onSuccess,
      onError: onError,
      config: config,
      isSponsored: isSponsored,
      title: title,
      experimental: experimental,
      headerLeftContent: backButton,
      children: [/*#__PURE__*/jsxs(SwapSettings, {
        className: cn('w-auto', classNames?.settings?.container),
        children: [/*#__PURE__*/jsx(SwapSettingsSlippageTitle, {
          className: classNames?.settings?.slippageTitle,
          children: "Max. slippage"
        }), /*#__PURE__*/jsx(SwapSettingsSlippageDescription, {
          className: classNames?.settings?.slippageDescription,
          children: "Your swap will revert if the prices change by more than the selected percentage."
        }), /*#__PURE__*/jsx(SwapSettingsSlippageInput, {
          className: classNames?.settings?.slippageInput
        })]
      }), /*#__PURE__*/jsx(SwapAmountInput, {
        label: "Sell",
        swappableTokens: from,
        type: "from",
        className: classNames?.fromAmountInput
      }), /*#__PURE__*/jsx(SwapToggleButton, {
        className: classNames?.toggleButton
      }), /*#__PURE__*/jsx(SwapAmountInput, {
        label: "Buy",
        swappableTokens: to,
        type: "to",
        className: classNames?.toAmountInput
      }), /*#__PURE__*/jsx(SwapButton, {
        disabled: disabled,
        className: classNames?.swapButton
      }), /*#__PURE__*/jsx(SwapMessage, {
        className: classNames?.message
      }), /*#__PURE__*/jsx(SwapToast, {
        className: classNames?.toast
      })]
    })
  });
}
export { WalletAdvancedSwap };
