// @coinbase/onchainkit/esm/wallet/components/WalletAdvancedWalletActions.js
'use client';
import { useAnalytics } from '../../core/analytics/hooks/useAnalytics.js';
import { WalletEvent, WalletOption } from '../../core/analytics/types.js';
import { PressableIcon } from '../../internal/components/PressableIcon.js';
import { baseScanSvg } from '../../internal/svg/baseScanSvg.js';
import { disconnectSvg } from '../../internal/svg/disconnectSvg.js';
import { qrIconSvg } from '../../internal/svg/qrIconSvg.js';
import { refreshSvg } from '../../internal/svg/refreshSvg.js';
import { cn } from '../../styles/theme.js';
import { useCallback } from 'react';
import { useAccount, useDisconnect } from 'wagmi';
import { useWalletContext } from './WalletProvider.js';
import { usePortfolio } from '../hooks/usePortfolio.js';
import { RequestContext } from '../../core/network/constants.js';
import { jsxs, jsx } from 'react/jsx-runtime';
function WalletAdvancedWalletActions({
  classNames
}) {
  const _useAccount = useAccount(),
    address = _useAccount.address;
  const _useWalletContext = useWalletContext(),
    handleClose = _useWalletContext.handleClose,
    setActiveFeature = _useWalletContext.setActiveFeature,
    animations = _useWalletContext.animations;
  const _useDisconnect = useDisconnect(),
    disconnect = _useDisconnect.disconnect,
    connectors = _useDisconnect.connectors;
  const _useAnalytics = useAnalytics(),
    sendAnalytics = _useAnalytics.sendAnalytics;
  const _usePortfolio = usePortfolio({
      address
    }, RequestContext.Wallet),
    refetchPortfolioData = _usePortfolio.refetch;
  const handleAnalyticsOptionSelected = useCallback(option => {
    sendAnalytics(WalletEvent.OptionSelected, {
      option
    });
  }, [sendAnalytics]);
  const handleAnalyticsDisconnect = useCallback(walletProvider => {
    sendAnalytics(WalletEvent.Disconnect, {
      component: 'WalletAdvanced',
      walletProvider
    });
  }, [sendAnalytics]);
  const handleTransactions = useCallback(() => {
    handleAnalyticsOptionSelected(WalletOption.Explorer);
    window.open(`https://basescan.org/address/${address}`, '_blank');
  }, [address, handleAnalyticsOptionSelected]);
  const handleDisconnect = useCallback(() => {
    const walletProvider = connectors[0]?.name || 'unknown';
    handleAnalyticsDisconnect(walletProvider);
    handleClose();
    for (const connector of connectors) {
      disconnect({
        connector
      });
    }
  }, [disconnect, connectors, handleClose, handleAnalyticsDisconnect]);
  const handleQr = useCallback(() => {
    handleAnalyticsOptionSelected(WalletOption.QR);
    setActiveFeature('qr');
  }, [setActiveFeature, handleAnalyticsOptionSelected]);
  const handleRefreshPortfolioData = useCallback(async () => {
    handleAnalyticsOptionSelected(WalletOption.Refresh);
    await refetchPortfolioData();
  }, [refetchPortfolioData, handleAnalyticsOptionSelected]);
  return /*#__PURE__*/jsxs("div", {
    "data-testid": "ockWalletAdvanced_WalletActions",
    className: cn('flex w-88 items-center justify-between px-4 py-3', animations.content, classNames?.container),
    children: [/*#__PURE__*/jsxs("div", {
      className: "flex items-center",
      children: [/*#__PURE__*/jsx(PressableIcon, {
        ariaLabel: "Open transaction history",
        onClick: handleTransactions,
        children: /*#__PURE__*/jsx("div", {
          "data-testid": "ockWalletAdvanced_TransactionsButton",
          className: cn('h-7 w-7 scale-110 p-2', classNames?.baseScanIcon),
          children: baseScanSvg
        })
      }), /*#__PURE__*/jsx(PressableIcon, {
        ariaLabel: "Show QR code",
        onClick: handleQr,
        children: /*#__PURE__*/jsx("div", {
          "data-testid": "ockWalletAdvanced_QrButton",
          className: cn('h-7 w-7 scale-110', classNames?.qrIcon),
          children: qrIconSvg
        })
      })]
    }), /*#__PURE__*/jsxs("div", {
      className: "flex items-center",
      children: [/*#__PURE__*/jsx(PressableIcon, {
        ariaLabel: "Disconnect wallet",
        onClick: handleDisconnect,
        children: /*#__PURE__*/jsx("div", {
          "data-testid": "ockWalletAdvanced_DisconnectButton",
          className: cn('h-7 w-7 scale-110 p-2', classNames?.disconnectIcon),
          children: disconnectSvg
        })
      }), /*#__PURE__*/jsx(PressableIcon, {
        ariaLabel: "Refresh portfolio data",
        onClick: handleRefreshPortfolioData,
        children: /*#__PURE__*/jsx("div", {
          "data-testid": "ockWalletAdvanced_RefreshButton",
          className: cn('h-7 w-7 scale-110 p-2', classNames?.refreshIcon),
          children: refreshSvg
        })
      })]
    })]
  });
}
export { WalletAdvancedWalletActions };
