// @coinbase/onchainkit/esm/wallet/components/WalletDropdown.js
'use client';
import '../../identity/index.js';
import { color, cn } from '../../styles/theme.js';
import { WalletDropdownContent } from './WalletDropdownContent.js';
import { WalletDropdownDisconnect } from './WalletDropdownDisconnect.js';
import { WalletDropdownLink } from './WalletDropdownLink.js';
import { useWalletContext } from './WalletProvider.js';
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { Identity } from '../../identity/components/Identity.js';
import { Avatar } from '../../identity/components/Avatar.js';
import { Name } from '../../identity/components/Name.js';
import { Address } from '../../identity/components/Address.js';
import { EthBalance } from '../../identity/components/EthBalance.js';
const defaultWalletDropdownChildren = /*#__PURE__*/jsxs(Fragment, {
  children: [/*#__PURE__*/jsxs(Identity, {
    className: "px-4 pt-3 pb-2",
    children: [/*#__PURE__*/jsx(Avatar, {}), /*#__PURE__*/jsx(Name, {}), /*#__PURE__*/jsx(Address, {
      className: color.foregroundMuted
    }), /*#__PURE__*/jsx(EthBalance, {})]
  }, "wallet-dd-identity"), /*#__PURE__*/jsx(WalletDropdownLink, {
    icon: "wallet",
    href: "https://keys.coinbase.com",
    target: "_blank",
    children: "Wallet"
  }, "wallet-dd-link"), /*#__PURE__*/jsx(WalletDropdownDisconnect, {}, "wallet-dd-disconnect")]
});
function WalletDropdown({
  children,
  className,
  classNames,
  swappableTokens
}) {
  const _useWalletContext = useWalletContext(),
    address = _useWalletContext.address,
    breakpoint = _useWalletContext.breakpoint,
    isSubComponentOpen = _useWalletContext.isSubComponentOpen,
    showSubComponentAbove = _useWalletContext.showSubComponentAbove,
    alignSubComponentRight = _useWalletContext.alignSubComponentRight;
  if (!address) {
    return null;
  }
  if (!breakpoint) {
    return null;
  }
  if (!isSubComponentOpen) {
    return null;
  }
  return /*#__PURE__*/jsx("div", {
    "data-testid": "ockWalletDropdown",
    className: cn('absolute', showSubComponentAbove ? 'bottom-full' : 'top-full', alignSubComponentRight ? 'right-0' : 'left-0', className),
    children: /*#__PURE__*/jsx(WalletDropdownContent, {
      classNames: classNames,
      swappableTokens: swappableTokens,
      children: children || defaultWalletDropdownChildren
    })
  });
}
export { WalletDropdown };
