// @coinbase/onchainkit/esm/wallet/components/WalletAdvancedAddressDetails.js
'use client';
function _slicedToArray(r, e) { return _arrayWithHoles(r) || _iterableToArrayLimit(r, e) || _unsupportedIterableToArray(r, e) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(r) { if (Array.isArray(r)) return r; }
import '../../identity/index.js';
import { Spinner } from '../../internal/components/Spinner.js';
import { zIndex } from '../../styles/constants.js';
import { cn, pressable, text, color, border } from '../../styles/theme.js';
import { useState, useCallback } from 'react';
import { useWalletContext } from './WalletProvider.js';
import { usePortfolio } from '../hooks/usePortfolio.js';
import { RequestContext } from '../../core/network/constants.js';
import { useAccount } from 'wagmi';
import { jsx, jsxs } from 'react/jsx-runtime';
import { Avatar } from '../../identity/components/Avatar.js';
import { Name } from '../../identity/components/Name.js';
function WalletAdvancedAddressDetails({
  classNames
}) {
  const _useWalletContext = useWalletContext(),
    address = _useWalletContext.address,
    chain = _useWalletContext.chain,
    animations = _useWalletContext.animations;
  const _useState = useState('Copy'),
    _useState2 = _slicedToArray(_useState, 2),
    copyText = _useState2[0],
    setCopyText = _useState2[1];
  const handleCopyAddress = useCallback(async () => {
    try {
      await navigator.clipboard.writeText(String(address));
      setCopyText('Copied');
      setTimeout(() => setCopyText('Copy'), 2000);
    } catch (err) {
      console.error('Failed to copy address:', err);
      setCopyText('Failed to copy');
      setTimeout(() => setCopyText('Copy'), 2000);
    }
  }, [address]);
  if (!address || !chain) {
    return /*#__PURE__*/jsx("div", {
      className: "mt-1 h-28 w-10 px-4 py-3"
    }); // Prevent layout shift
  }
  return /*#__PURE__*/jsxs("div", {
    "data-testid": "ockWalletAdvanced_AddressDetails",
    className: cn('mt-2 flex w-88 flex-col items-center justify-center px-4 py-3', color.foreground, text.body, animations.content, classNames?.container),
    children: [/*#__PURE__*/jsx(Avatar, {
      address: address,
      chain: chain,
      className: cn('pointer-events-none h-10 w-10', classNames?.avatar)
    }), /*#__PURE__*/jsxs("div", {
      className: "group relative mt-2 text-base",
      children: [/*#__PURE__*/jsx("button", {
        type: "button",
        onClick: handleCopyAddress,
        "data-testid": "ockWalletAdvanced_NameButton",
        children: /*#__PURE__*/jsx(Name, {
          address: address,
          chain: chain,
          className: cn('hover:text-[var(--ock-text-foreground-muted)] active:text-[var(--ock-text-primary)]', classNames?.nameButton)
        })
      }), /*#__PURE__*/jsx("button", {
        type: "button",
        onClick: handleCopyAddress,
        className: cn(pressable.alternate, text.legal, color.foreground, border.default, border.radius, zIndex.tooltip, 'absolute top-full right-0 mt-0.5 px-1.5 py-0.5 opacity-0 transition-opacity group-hover:opacity-100'),
        "aria-live": "polite",
        "data-testid": "ockWalletAdvanced_NameTooltip",
        children: copyText
      })]
    }), /*#__PURE__*/jsx(AddressBalanceInFiat, {
      className: classNames?.fiatBalance
    })]
  });
}
function AddressBalanceInFiat({
  className
}) {
  const _useAccount = useAccount(),
    address = _useAccount.address;
  const _usePortfolio = usePortfolio({
      address
    }, RequestContext.Wallet),
    portfolioData = _usePortfolio.data,
    isFetchingPortfolioData = _usePortfolio.isFetching;
  const portfolioFiatValue = portfolioData?.portfolioBalanceInUsd;
  const formattedValueInFiat = new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD'
  }).format(Number(portfolioFiatValue));
  if (isFetchingPortfolioData) {
    return /*#__PURE__*/jsx("div", {
      className: "mt-1 h-8 w-full",
      children: /*#__PURE__*/jsx(Spinner, {})
    });
  }
  if (portfolioFiatValue === null || portfolioFiatValue === undefined) {
    return null;
  }
  return /*#__PURE__*/jsx("div", {
    className: cn(text.title1, 'mt-1 font-normal', className),
    "data-testid": "ockWalletAdvanced_AddressBalance",
    children: formattedValueInFiat
  });
}
export { WalletAdvancedAddressDetails };
