// @coinbase/onchainkit/esm/wallet/components/WalletAdvancedTokenHoldings.js
'use client';
import { cn, text, color } from '../../styles/theme.js';
import '../../token/index.js';
import { formatUnits } from 'viem';
import { useWalletContext } from './WalletProvider.js';
import { usePortfolio } from '../hooks/usePortfolio.js';
import { RequestContext } from '../../core/network/constants.js';
import { useAccount } from 'wagmi';
import { jsx, jsxs } from 'react/jsx-runtime';
import { TokenImage } from '../../token/components/TokenImage.js';
function WalletAdvancedTokenHoldings({
  classNames
}) {
  const _useAccount = useAccount(),
    address = _useAccount.address;
  const _useWalletContext = useWalletContext(),
    animations = _useWalletContext.animations;
  const _usePortfolio = usePortfolio({
      address
    }, RequestContext.Wallet),
    portfolioData = _usePortfolio.data,
    isFetchingPortfolioData = _usePortfolio.isFetching;
  const tokenBalances = portfolioData?.tokenBalances;
  if (isFetchingPortfolioData || !tokenBalances || tokenBalances.length === 0) {
    return /*#__PURE__*/jsx("div", {
      "data-testid": "ockWalletAdvanced_LoadingPlaceholder",
      className: "mx-auto my-2 h-44 w-80 px-4 py-3"
    }); // Prevent layout shift
  }
  return /*#__PURE__*/jsx("div", {
    className: cn('flex flex-col items-center gap-4', 'my-2 h-44 max-h-44', 'scrollbar-hidden w-88 overflow-y-auto px-4 py-3', animations.content, classNames?.container),
    "data-testid": "ockWalletAdvanced_TokenHoldings",
    children: tokenBalances.map((tokenBalance, index) => /*#__PURE__*/jsx(TokenDetails, {
      token: {
        address: tokenBalance.address,
        chainId: tokenBalance.chainId,
        decimals: tokenBalance.decimals,
        image: tokenBalance.image,
        name: tokenBalance.name,
        symbol: tokenBalance.symbol
      },
      balance: Number(formatUnits(BigInt(tokenBalance.cryptoBalance), tokenBalance.decimals)),
      valueInFiat: Number(tokenBalance.fiatBalance),
      classNames: classNames?.tokenDetails
    }, `${tokenBalance.address}-${index}`))
  });
}
function TokenDetails({
  token,
  balance,
  valueInFiat,
  classNames,
  tokenImageSize = 32
}) {
  const formattedBalance = new Intl.NumberFormat('en-US', {
    minimumFractionDigits: 2,
    maximumFractionDigits: 5
  }).format(balance);
  const formattedValueInFiat = new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD'
  }).format(valueInFiat);
  return /*#__PURE__*/jsxs("div", {
    className: cn('flex w-full flex-row items-center justify-between', classNames?.container),
    children: [/*#__PURE__*/jsxs("div", {
      "data-testid": "ockWalletAdvanced_TokenDetails_TokenImage",
      className: cn('flex flex-row items-center gap-2', classNames?.tokenImage),
      children: [/*#__PURE__*/jsx(TokenImage, {
        token: token,
        size: tokenImageSize
      }), /*#__PURE__*/jsxs("div", {
        className: "flex flex-col",
        children: [/*#__PURE__*/jsx("span", {
          className: cn(text.label1, color.foreground, 'max-w-52 overflow-hidden text-ellipsis whitespace-nowrap text-left', classNames?.tokenName),
          children: token.name?.trim()
        }), /*#__PURE__*/jsx("span", {
          className: cn(text.legal, color.foregroundMuted, classNames?.tokenBalance),
          children: `${formattedBalance} ${token.symbol}`
        })]
      })]
    }), /*#__PURE__*/jsx("span", {
      className: cn(text.label2, color.foregroundMuted, classNames?.fiatValue),
      children: formattedValueInFiat
    })]
  });
}
export { WalletAdvancedTokenHoldings };
