// @coinbase/onchainkit/esm/wallet/components/WalletIsland.js
'use client';
import '../../identity/index.js';
import { portfolioSvg } from '../../internal/svg/portfolioSvg.js';
import { getDefaultDraggableStartingPosition } from '../utils/getDefaultDraggableStartingPosition.js';
import { ConnectWallet } from './ConnectWallet.js';
import { Wallet } from './Wallet.js';
import { WalletAdvancedAddressDetails } from './WalletAdvancedAddressDetails.js';
import { WalletAdvancedTokenHoldings } from './WalletAdvancedTokenHoldings.js';
import { WalletAdvancedTransactionActions } from './WalletAdvancedTransactionActions.js';
import { WalletAdvancedWalletActions } from './WalletAdvancedWalletActions.js';
import { WalletDropdown } from './WalletDropdown.js';
import { jsxs, jsx } from 'react/jsx-runtime';
import { Avatar } from '../../identity/components/Avatar.js';
function WalletIsland({
  startingPosition = getDefaultDraggableStartingPosition()
}) {
  return /*#__PURE__*/jsxs(Wallet, {
    draggable: true,
    draggableStartingPosition: startingPosition,
    children: [/*#__PURE__*/jsx(ConnectWallet, {
      className: "!rounded-full m-0 flex h-14 w-14 min-w-14 flex-col items-center justify-center p-0",
      disconnectedLabel: /*#__PURE__*/jsx("div", {
        className: "h-5 w-5",
        children: portfolioSvg
      }),
      children: /*#__PURE__*/jsx(Avatar, {
        className: "pointer-events-none h-14 w-14"
      })
    }), /*#__PURE__*/jsxs(WalletDropdown, {
      children: [/*#__PURE__*/jsx(WalletAdvancedWalletActions, {}), /*#__PURE__*/jsx(WalletAdvancedAddressDetails, {}), /*#__PURE__*/jsx(WalletAdvancedTransactionActions, {}), /*#__PURE__*/jsx(WalletAdvancedTokenHoldings, {})]
    })]
  });
}
export { WalletIsland };
