// @coinbase/onchainkit/esm/transaction/components/Transaction.js
import { useIsMounted } from '../../internal/hooks/useIsMounted.js';
import { useTheme } from '../../internal/hooks/useTheme.js';
import { cn } from '../../styles/theme.js';
import { useOnchainKit } from '../../useOnchainKit.js';
import { TransactionButton } from './TransactionButton.js';
import { TransactionProvider } from './TransactionProvider.js';
import { TransactionToast } from './TransactionToast.js';
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
function Transaction({
  calls,
  capabilities,
  chainId,
  className,
  children,
  contracts,
  disabled = false,
  isSponsored,
  onError,
  onStatus,
  onSuccess,
  resetAfter
}) {
  const isMounted = useIsMounted();
  const componentTheme = useTheme();
  const _useOnchainKit = useOnchainKit(),
    chain = _useOnchainKit.chain;

  // prevents SSR hydration issue
  if (!isMounted) {
    return /*#__PURE__*/jsx("div", {
      className: cn(componentTheme, 'flex w-full flex-col gap-2', className)
    });
  }

  // If chainId is not provided,
  // use the default chainId from the OnchainKit context
  const accountChainId = chainId ? chainId : chain.id;
  return /*#__PURE__*/jsx(TransactionProvider, {
    calls: calls,
    capabilities: capabilities,
    chainId: accountChainId,
    contracts: contracts,
    isSponsored: isSponsored,
    onError: onError,
    onStatus: onStatus,
    onSuccess: onSuccess,
    resetAfter: resetAfter,
    children: /*#__PURE__*/jsx("div", {
      className: cn(componentTheme, 'flex w-full flex-col gap-2', className),
      children: children ?? /*#__PURE__*/jsxs(Fragment, {
        children: [/*#__PURE__*/jsx(TransactionButton, {
          disabled: disabled
        }), /*#__PURE__*/jsx(TransactionToast, {})]
      })
    })
  });
}
export { Transaction };
