// @coinbase/onchainkit/esm/buy/components/BuyAmountInput.js
'use client';
import { TextInput } from '../../internal/components/TextInput.js';
import { isValidAmount } from '../../internal/utils/isValidAmount.js';
import { cn, background, color, border } from '../../styles/theme.js';
import { formatAmount } from '../../swap/utils/formatAmount.js';
import '../../token/index.js';
import { useBuyContext } from './BuyProvider.js';
import { jsxs, jsx } from 'react/jsx-runtime';
import { TokenChip } from '../../token/components/TokenChip.js';
function BuyAmountInput() {
  const _useBuyContext = useBuyContext(),
    to = _useBuyContext.to,
    handleAmountChange = _useBuyContext.handleAmountChange;
  if (!to?.token) {
    return null;
  }
  return /*#__PURE__*/jsxs("div", {
    className: cn('flex h-12 items-center border px-2 pl-4', background.default, border.radius, border.lineDefault),
    children: [/*#__PURE__*/jsx(TextInput, {
      className: cn('mr-2 w-full border-none font-display', 'leading-none outline-none disabled:cursor-not-allowed', background.default, color.foreground),
      placeholder: "0.0",
      delayMs: 1000,
      inputMode: "decimal",
      value: formatAmount(to.amount),
      setValue: to.setAmount,
      disabled: to.loading,
      onChange: handleAmountChange,
      inputValidator: isValidAmount
    }), /*#__PURE__*/jsx(TokenChip, {
      className: cn(color.foreground, 'rounded-md'),
      token: to.token,
      isPressable: false
    })]
  });
}
export { BuyAmountInput };
