// @coinbase/onchainkit/esm/buy/components/BuyOnrampItem.js
'use client';
import { usdcToken } from '../../token/constants.js';
import { useCallback, useMemo } from 'react';
import { applePaySvg } from '../../internal/svg/applePaySvg.js';
import { cardSvg } from '../../internal/svg/cardSvg.js';
import { coinbaseLogoSvg } from '../../internal/svg/coinbaseLogoSvg.js';
import { cn, color, text, pressable } from '../../styles/theme.js';
import { useBuyContext } from './BuyProvider.js';
import { jsxs, jsx } from 'react/jsx-runtime';
const ONRAMP_ICON_MAP = {
  applePay: applePaySvg,
  coinbasePay: coinbaseLogoSvg,
  creditCard: cardSvg
};
function BuyOnrampItem({
  name,
  description,
  onClick,
  icon,
  to
}) {
  const _useBuyContext = useBuyContext(),
    setIsDropdownOpen = _useBuyContext.setIsDropdownOpen;
  const handleClick = useCallback(() => {
    setIsDropdownOpen(false);
    onClick();
  }, [onClick, setIsDropdownOpen]);
  const fiatAmount = useMemo(() => {
    // if token is USDC the usd estimate
    // can be slightly off (4.9999999) so
    // use amount instead to prevent disabling of onramp
    if (to?.token?.address?.toLowerCase() === usdcToken?.address.toLowerCase()) {
      return to?.amount;
    }
    return to?.amountUSD;
  }, [to]);
  // Debit and Apple Pay have a minimum purchase amount of $5
  const isDisabled = !fiatAmount || Number.parseFloat(fiatAmount) < 5 && name !== 'Coinbase';
  const message = useMemo(() => {
    if (isDisabled) {
      return 'Minimum purchase amount is $5';
    }
    return description;
  }, [isDisabled, description]);
  return /*#__PURE__*/jsxs("button", {
    className: cn('flex items-center gap-2 rounded-lg p-2', text.label2, !isDisabled && pressable.default, isDisabled && color.foregroundMuted),
    onClick: handleClick,
    type: "button",
    "data-testid": `ock-${icon}OnrampItem`,
    disabled: isDisabled,
    children: [/*#__PURE__*/jsx("div", {
      className: "flex h-9 w-9 items-center justify-center",
      children: ONRAMP_ICON_MAP[icon]
    }), /*#__PURE__*/jsxs("div", {
      className: "flex flex-col items-start",
      children: [/*#__PURE__*/jsx("div", {
        className: "relative flex items-center gap-1",
        children: /*#__PURE__*/jsx("div", {
          children: name
        })
      }), /*#__PURE__*/jsx("div", {
        className: cn('text-xs', color.foregroundMuted),
        children: message
      })]
    })]
  });
}
export { BuyOnrampItem };
