// @coinbase/onchainkit/esm/wallet/components/wallet-advanced-send/components/SendAmountInput.js
'use client';
import { AmountInput } from '../../../../internal/components/amount-input/AmountInput.js';
import { SendAmountInputTypeSwitch } from './SendAmountInputTypeSwitch.js';
import { useSendContext } from './SendProvider.js';
import { jsxs, jsx } from 'react/jsx-runtime';
function SendAmountInput({
  className,
  textClassName
}) {
  const _useSendContext = useSendContext(),
    selectedToken = _useSendContext.selectedToken,
    cryptoAmount = _useSendContext.cryptoAmount,
    handleCryptoAmountChange = _useSendContext.handleCryptoAmountChange,
    fiatAmount = _useSendContext.fiatAmount,
    handleFiatAmountChange = _useSendContext.handleFiatAmountChange,
    selectedInputType = _useSendContext.selectedInputType,
    exchangeRate = _useSendContext.exchangeRate;
  return /*#__PURE__*/jsxs("div", {
    className: "flex h-full w-full flex-col justify-center",
    children: [/*#__PURE__*/jsx(AmountInput, {
      fiatAmount: fiatAmount ?? '',
      cryptoAmount: cryptoAmount ?? '',
      asset: selectedToken?.symbol ?? '',
      currency: "USD",
      selectedInputType: selectedInputType,
      setFiatAmount: handleFiatAmountChange,
      setCryptoAmount: handleCryptoAmountChange,
      exchangeRate: String(exchangeRate),
      className: className,
      textClassName: textClassName
    }), /*#__PURE__*/jsx(SendAmountInputTypeSwitch, {})]
  });
}
export { SendAmountInput };
