// @coinbase/onchainkit/esm/wallet/components/wallet-advanced-send/components/SendAmountInputTypeSwitch.js
import { Skeleton } from '../../../../internal/components/Skeleton.js';
import { AmountInputTypeSwitch } from '../../../../internal/components/amount-input/AmountInputTypeSwitch.js';
import { cn, text, color } from '../../../../styles/theme.js';
import { useSendContext } from './SendProvider.js';
import { jsx } from 'react/jsx-runtime';
const defaultErrorDisplay = /*#__PURE__*/jsx("div", {
  "data-testid": "ockSendAmountInputTypeSwitch_ErrorDisplay",
  className: cn(text.caption, color.foregroundMuted, 'h-[1.625rem]'),
  children: "Exchange rate unavailable"
});
function SendAmountInputTypeSwitch({
  errorDisplay,
  className
}) {
  const _useSendContext = useSendContext(),
    selectedToken = _useSendContext.selectedToken,
    fiatAmount = _useSendContext.fiatAmount,
    cryptoAmount = _useSendContext.cryptoAmount,
    exchangeRate = _useSendContext.exchangeRate,
    exchangeRateLoading = _useSendContext.exchangeRateLoading,
    selectedInputType = _useSendContext.selectedInputType,
    setSelectedInputType = _useSendContext.setSelectedInputType;
  if (exchangeRateLoading) {
    return /*#__PURE__*/jsx(Skeleton, {
      className: "h-[1.625rem]"
    });
  }
  if (!exchangeRate) {
    if (errorDisplay) {
      return errorDisplay;
    }
    return defaultErrorDisplay;
  }
  return /*#__PURE__*/jsx(AmountInputTypeSwitch, {
    asset: selectedToken?.symbol ?? '',
    fiatAmount: fiatAmount ?? '',
    cryptoAmount: cryptoAmount ?? '',
    exchangeRate: exchangeRate,
    exchangeRateLoading: false,
    currency: "USD",
    selectedInputType: selectedInputType,
    setSelectedInputType: setSelectedInputType,
    className: className
  });
}
export { SendAmountInputTypeSwitch };
