// @coinbase/onchainkit/esm/wallet/components/wallet-advanced-send/components/SendAddressSelector.js
'use client';
import '../../../../identity/index.js';
import { cn, background, border, pressable } from '../../../../styles/theme.js';
import { jsx, jsxs } from 'react/jsx-runtime';
import { Avatar } from '../../../../identity/components/Avatar.js';
import { Name } from '../../../../identity/components/Name.js';
import { Address } from '../../../../identity/components/Address.js';
function SendAddressSelector({
  address,
  senderChain,
  onClick,
  classNames
}) {
  if (!address || !senderChain) {
    return null;
  }
  return /*#__PURE__*/jsx("button", {
    "data-testid": "ockSendAddressSelector_button",
    type: "button",
    onClick: onClick,
    className: "w-full text-left",
    children: /*#__PURE__*/jsx("div", {
      "data-testid": "ockSendAddressSelector_container",
      className: cn(background.default, border.radius, pressable.default, 'items-left flex min-w-[300px]', 'mt-2 p-2', classNames?.container),
      children: /*#__PURE__*/jsxs("div", {
        className: "flex items-center space-x-3",
        children: [/*#__PURE__*/jsx("div", {
          className: "flex-shrink-0",
          children: /*#__PURE__*/jsx(Avatar, {
            address: address,
            chain: senderChain,
            className: classNames?.avatar
          })
        }), /*#__PURE__*/jsxs("div", {
          className: "flex flex-col",
          children: [/*#__PURE__*/jsx(Name, {
            address: address,
            chain: senderChain,
            className: classNames?.name
          }), /*#__PURE__*/jsx(Address, {
            address: address,
            hasCopyAddressOnClick: false,
            className: classNames?.address
          })]
        })]
      })
    })
  });
}
export { SendAddressSelector };
