// @coinbase/onchainkit/esm/internal/components/TextInput.js
import { forwardRef, useCallback } from 'react';
import { useDebounce } from '../hooks/useDebounce.js';
import { jsx } from 'react/jsx-runtime';
const TextInput = /*#__PURE__*/forwardRef(({
  'aria-label': ariaLabel,
  className,
  delayMs = 0,
  disabled = false,
  onBlur,
  onChange,
  onFocus,
  placeholder,
  setValue,
  inputMode,
  value,
  inputValidator = () => true
}, ref) => {
  const handleDebounce = useDebounce(value => {
    onChange(value);
  }, delayMs);
  const handleChange = useCallback(evt => {
    const value = evt.target.value;
    if (inputValidator(value)) {
      setValue?.(value);
      if (delayMs > 0) {
        handleDebounce(value);
      } else {
        onChange(value);
      }
    }
  }, [onChange, handleDebounce, delayMs, setValue, inputValidator]);
  return /*#__PURE__*/jsx("input", {
    "aria-label": ariaLabel,
    "data-testid": "ockTextInput_Input",
    ref: ref,
    type: "text",
    className: className,
    inputMode: inputMode,
    placeholder: placeholder,
    value: value,
    onBlur: onBlur,
    onChange: handleChange,
    onFocus: onFocus,
    disabled: disabled,
    autoComplete: "off" // autocomplete attribute handles browser autocomplete
    ,

    "data-1p-ignore": true // data-1p-ignore attribute handles password manager autocomplete
  });
});
TextInput.displayName = 'TextInput';
export { TextInput };
