// @coinbase/onchainkit/esm/fund/components/FundCardPaymentMethodSelectorToggle.js
import { forwardRef } from 'react';
import { caretUpSvg } from '../../internal/svg/caretUpSvg.js';
import { cn, text, color, pressable, border } from '../../styles/theme.js';
import { FundCardPaymentMethodImage } from './FundCardPaymentMethodImage.js';
import { jsxs, jsx } from 'react/jsx-runtime';
const FundCardPaymentMethodSelectorToggle = /*#__PURE__*/forwardRef(({
  onClick,
  paymentMethod,
  isOpen,
  className
}, ref) => {
  return /*#__PURE__*/jsxs("button", {
    type: "button",
    className: cn(pressable.default, border.radius, border.lineDefault, 'flex h-12 w-full items-center gap-2 px-3 py-1', className),
    onClick: onClick,
    ref: ref,
    "data-testid": "ockFundCardPaymentMethodSelectorToggle",
    children: [/*#__PURE__*/jsx("div", {
      className: "w-4",
      children: /*#__PURE__*/jsx(FundCardPaymentMethodImage, {
        paymentMethod: paymentMethod,
        className: "h-4 w-4"
      })
    }), /*#__PURE__*/jsx("span", {
      className: cn(text.headline, color.foreground, 'flex w-full'),
      "data-testid": "ockFundCardPaymentMethodSelectorToggle__paymentMethodName",
      children: paymentMethod.name
    }), /*#__PURE__*/jsx("span", {
      className: cn('rotate-90 transition-transform duration-200', isOpen && 'rotate-180'),
      children: caretUpSvg
    })]
  });
});
FundCardPaymentMethodSelectorToggle.displayName = 'FundCardPaymentMethodSelectorToggle';
export { FundCardPaymentMethodSelectorToggle };
