// @coinbase/onchainkit/esm/fund/components/FundCardPresetAmountInputList.js
'use client';
import { useAmountInput } from '../../internal/hooks/useAmountInput.js';
import { FundCardPresetAmountInputItem } from './FundCardPresetAmountInputItem.js';
import { useFundContext } from './FundCardProvider.js';
import { jsx } from 'react/jsx-runtime';
function FundCardPresetAmountInputList() {
  const _useFundContext = useFundContext(),
    presetAmountInputs = _useFundContext.presetAmountInputs,
    currency = _useFundContext.currency,
    selectedInputType = _useFundContext.selectedInputType,
    exchangeRate = _useFundContext.exchangeRate,
    setFundAmountFiat = _useFundContext.setFundAmountFiat,
    setFundAmountCrypto = _useFundContext.setFundAmountCrypto;
  const _useAmountInput = useAmountInput({
      setFiatAmount: setFundAmountFiat,
      setCryptoAmount: setFundAmountCrypto,
      selectedInputType,
      exchangeRate: String(exchangeRate)
    }),
    handleFiatChange = _useAmountInput.handleFiatChange;
  if (!presetAmountInputs) {
    return null;
  }
  return /*#__PURE__*/jsx("div", {
    "data-testid": "ockPresetAmountInputList",
    className: "flex w-full flex-wrap items-center justify-between gap-2 pt-8",
    children: presetAmountInputs.map((amount, index) => /*#__PURE__*/jsx(FundCardPresetAmountInputItem, {
      presetAmountInput: amount,
      onClick: handleFiatChange,
      currency: currency
    }, index))
  });
}
export { FundCardPresetAmountInputList };
