// @coinbase/onchainkit/esm/fund/components/FundCardAmountInputTypeSwitch.js
'use client';
import { AmountInputTypeSwitch } from '../../internal/components/amount-input/AmountInputTypeSwitch.js';
import { useFundContext } from './FundCardProvider.js';
import { jsx } from 'react/jsx-runtime';
const FundCardAmountInputTypeSwitch = ({
  className
}) => {
  const _useFundContext = useFundContext(),
    selectedInputType = _useFundContext.selectedInputType,
    setSelectedInputType = _useFundContext.setSelectedInputType,
    asset = _useFundContext.asset,
    fundAmountFiat = _useFundContext.fundAmountFiat,
    fundAmountCrypto = _useFundContext.fundAmountCrypto,
    exchangeRate = _useFundContext.exchangeRate,
    exchangeRateLoading = _useFundContext.exchangeRateLoading,
    currency = _useFundContext.currency;
  return /*#__PURE__*/jsx(AmountInputTypeSwitch, {
    selectedInputType: selectedInputType,
    setSelectedInputType: setSelectedInputType,
    asset: asset,
    fiatAmount: fundAmountFiat,
    cryptoAmount: fundAmountCrypto,
    exchangeRate: exchangeRate,
    exchangeRateLoading: exchangeRateLoading,
    currency: currency,
    className: className
  });
};
export { FundCardAmountInputTypeSwitch, FundCardAmountInputTypeSwitch as default };
